e2e-tests/src/waiting.rs

```diff
@@ -68,3 +66,42 @@ pub fn wait_for_finalized_block(connection: Connection, block_number: u32) -> an
 
     Err(anyhow::anyhow!("Giving up"))
 }
+
+/// blocks the main thread waiting for an approval for proposal with id `proposal_id`
+pub fn wait_for_approval(connection: &Connection, proposal_id: u32) -> anyhow::Result<()> {
```

Comment:
One of these things is not like the others... is there no "approval" event?

Reply:
answered in another conversation